sources/packages/web/src/components/common/students/StudentProfileLegacyMatches.vue

```diff
@@ -0,0 +1,200 @@
+          Check for legacy matches
+          <v-tooltip activator="parent" location="start"
+            >Click to check for potential legacy matches.</v-tooltip
+          >
+        </v-btn></template
+      >
+    </banner>
+    <toggle-content
+      v-if="showLegacyMatches"
+      :toggled="!legacyStudentMatches?.length && !studentLegacyMatchesLoading"
+      message="No potential legacy matches found."
+    >
+      <v-data-table
+        :headers="StudentProfileLegacyMatchHeaders"
+        :items="legacyStudentMatches"
+        :loading="studentLegacyMatchesLoading"
+        :items-per-page="-1"
+      >
+        <template v-slot:loading>
+          <v-skeleton-loader type="table-row@3"></v-skeleton-loader>
+        </template>
+        <template #[`item.firstName`]="{ item }">
+          {{ emptyStringFiller(item.firstName) }}
+        </template>
+        <template #[`item.lastName`]="{ item }">
+          {{ item.lastName }}
+        </template>
+        <template #[`item.birthDate`]="{ item }">
+          {{ dateOnlyLongString(item.birthDate) }}
+        </template>
+        <template #[`item.sin`]="{ item }">
+          {{ sinDisplayFormat(item.sin) }}
+        </template>
+        <template #[`item.actions`]="{ item }">
+          <check-permission-role :role="Role.StudentLinkLegacyProfile">
+            <template #="{ notAllowed }">
+              <v-btn
+                color="primary"
+                :disabled="notAllowed"
+                @click="showLinkProfileModal(item.individualId)"
+              >
+                Link
+                <v-tooltip activator="parent" location="start"
+                  >Click to link student with this legacy profile.</v-tooltip
+                >
+              </v-btn>
+            </template>
+          </check-permission-role>
+        </template>
+        <template #bottom></template>
+      </v-data-table>
+    </toggle-content>
+    <student-profile-legacy-matches-modal ref="legacyMatchesModal" />
+  </template>
+</template>
+
+<script lang="ts">
+import { ref, defineComponent } from "vue";
+import { StudentService } from "@/services/StudentService";
+import { ModalDialog, useFormatters, useSnackBar } from "@/composables";
+import {
+  LegacyStudentMatchAPIOutDTO,
+  LegacyStudentMatchesAPIInDTO,
+  LegacyStudentProfileAPIOutDTO,
+} from "@/services/http/dto";
+import { Role, BannerTypes, StudentProfileLegacyMatchHeaders } from "@/types";
+import CheckPermissionRole from "@/components/generic/CheckPermissionRole.vue";
+import StudentProfileLegacyMatchesModal from "@/components/aest/students/modals/StudentProfileLegacyMatchesModal.vue";
+
+export default defineComponent({
+  emits: {
+    legacyProfileLinked: null,
+  },
+  components: {
+    CheckPermissionRole,
+    StudentProfileLegacyMatchesModal,
+  },
+  props: {
+    studentId: {
+      type: Number,
+      required: true,
+    },
+    legacyProfile: {
+      type: Object as () => LegacyStudentProfileAPIOutDTO,
```

Comment:
In other places we have been defining the property type using. `type: Object as PropType<LegacyStudentProfileAPIOutDTO>` is there a reason to use this way?

Reply:
Added by mistake, adjusted as recommended.